app/src/main/java/sopt/org/millie/presentation/search/SearchScreen.kt

```diff
@@ -0,0 +1,181 @@
+package sopt.org.millie.presentation.search
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import sopt.org.millie.R
+import sopt.org.millie.core.designsystem.component.MillieSearchTextField
+import sopt.org.millie.core.designsystem.component.MillieTabbar
+import sopt.org.millie.core.designsystem.component.MillieTopappbar
+import sopt.org.millie.core.designsystem.theme.MillieTheme
+import sopt.org.millie.presentation.search.book.SearchBookScreen
+import sopt.org.millie.presentation.search.library.SearchLibraryScreen
+import sopt.org.millie.presentation.search.model.SearchBannerModel
+import sopt.org.millie.presentation.search.model.SearchBookModel
+import sopt.org.millie.presentation.search.model.SearchLibraryModel
+
+@Composable
+fun SearchScreen(
+    searchBookList: List<SearchBookModel>,
+    searchLibraryList: List<SearchLibraryModel>,
+    searchBanner: SearchBannerModel,
+    onClickBookItem: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var text by remember { mutableStateOf("") }
+
+    val searchTabs = listOf("도서", "밀리로드", "포스트", "서재")
+    var selectedTabs by remember { mutableStateOf("도서") }
```

Comment:
SearchScreen 내부에서 text와 selectedTabs 상태를 직접 가지고 있으면 상위 화면에서 검색어 변경이나 탭 전환을 제어하기 어렵고 테스트나 재사용성도 떨어진답니동 !! 컴포즈에서는 이런 UI 상태를 상위에서 관리하고 Composable은 UI만 그리도록 하는 상태 호이스팅을 권장하고 있어서 이 두 상태를 함수 파라미터로 올려서 관리하는 게 더 적절해 보일 것 같은데 어떻게 생각시나요 ?!

Reply:
그리구 선택된 탭이라면 복수형 보다 단수형이 더 나을듯 !!
```suggestion
    var text by remember { mutableStateOf("") }

    val searchTabs = listOf("도서", "밀리로드", "포스트", "서재")
    var selectedTab by remember { mutableStateOf("도서") }
```

Reply:
이 화면에서만 쓰이는 상태라고 생각해서 바로 사용했는데 screen을 ui만 그리는 stateless한 컴포저블로 만들기 위해서도 호이스팅을 해주는군요..! 그럼 screen에서 사용되는 모든 상태는 호이스팅해주는게 좋은가요?

Reply:
모든 상태를 무조건 호이스팅할 필요는 없어요.
상태 호이스팅의 핵심 목적은 Composable을 Stateless하게 만들어 UI와 상태를 분리하고, 재사용성과 테스트 용이성을 높이는 데 있으니까용 그런데 보통은 화면에서만 쓰이는 상태라도 가능하면 호이스팅해서 Stateless하게 만드는 것이 좋아용 Stateless하게 구성하면 UI와 상태가 분리되어 재사용성이 높아지고 테스트하기도 쉽고 Compose 리컴포지션 최적화에도 유리하답니당 !! 화면이 단순히 데이터를 받아서 그리는 역할만 하도록 만들면 유지보수와 확장성 측면에서도 장점이 크구요 !